app/code/community/Algolia/Algoliasearch/Helper/Entity/Producthelper.php

```diff
@@ -512,102 +521,100 @@ protected function handlePrice(Mage_Catalog_Model_Product &$product, $sub_produc
             $weeeTaxAmount = 0;
         }
 
-        $baseCurrencyCode = $store->getBaseCurrencyCode();
-
-        $groups = array();
-
         if ($customer_groups_enabled) {
             $groups = Mage::getModel('customer/group')->getCollection();
+        } else {
+            $groups = array();
+        }
+        $special_price = $product->getSpecialPrice();
+        $special_from_date = $product->getSpecialFromDate();
+        $special_to_date = $product->getSpecialToDate();
+
+        // There is a special_price and either no to date or a to date in the future
+        // start date is irrelevant
+        if ($special_price && Mage::app()->getLocale()->isStoreDateInInterval($store, null, $special_to_date)) {
+            $canSpecialBeValid = true;
+            $special_price += $weeeTaxAmount;
         }
-
-        /** @var Mage_Tax_Helper_Data $taxHelper */
-        $taxHelper = Mage::helper('tax');
-
-        /** @var Mage_Directory_Helper_Data $directoryHelper */
-        $directoryHelper = Mage::helper('directory');
 
         foreach ($fields as $field => $with_tax) {
             $customData[$field] = array();
+            $field_price = (double) $taxHelper->getPrice($product, $product->getPrice() + $weeeTaxAmount, $with_tax, null, null, null, $store, null);
+            if ($canSpecialBeValid) {
+                $field_special_price = (double) $taxHelper->getPrice($product, $special_price, $with_tax, null, null, null, $store, null);
+            }
 
             foreach ($currencies as $currency_code) {
-                $customData[$field][$currency_code] = array();
+                $cdfc = array();
```

Comment:
I found it really hard to read  $customData[$field][$currency_code] everywhere, so I renamed this and added it at the end, I apologise if this breaks your code style

Reply:
You're right, this is painful to use in the rest of the function. On the other hand $cdfc isn't very meaningful and maybe too short. Could you please consider to change the variable name in something meaningful like $currencyData maybe ? Thx in advance :)